library/gensym.py

```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/python
+
+import subprocess
+from ansible.module_utils.basic import AnsibleModule 
+
+def test_name(name, used_list):
+    for item in used_list:
+        if name in item:
+            return False
+    return True
```

Comment:
This `test_name` function will return False if any of the names in `used_list` contain `name`, but we only care if one of the names in `used_list` _is_ `name`. Consider a proposed name of 'foo' and a `used_list` of `['food', 'bar']` -- you would want the test to pass for 'foo' in this case. You could check this with `if name not in used_list: ...`